tests/unit/test_logging.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""Tests for the log module."""
+
+import json
+import logging
+from contextlib import contextmanager
+from dataclasses import dataclass, field
+
+import pytest
+
+from hexkit.log import (
+    JsonFormatter,
+    LoggingConfig,
+    RecordCompiler,
+    configure_logging,
+)
+
+VALID_SERVICE_NAME = "test"
+VALID_INSTANCE_ID = "001"
+VALID_CONFIG = LoggingConfig(
+    log_level="INFO",
+    service_name="hexkit-test",
+    service_instance_id="0",
+)
+
+DEFAULT_KEYS = (  # all the keys present in the resulting log
```

Comment:
You could use the RESERVED_RECORD_KEYS from the log module instead.

Reply:
The DEFAULT_KEYS lists the final fields included in the JSON log, which are different from the built-in attributes found in RESERVED_RECORD_KEYS. Maybe it should be renamed for clarity

Reply:
Ah OK, sorry missunderstood